test/integration/test_mail.py

```diff
@@ -0,0 +1,166 @@
+import pytest
+import requests
+from typing import List
+from testcontainers.core.container import DockerContainer
+from autosubmit.notifications.mail_notifier import MailNotifier
+from autosubmit.job.job_common import Status
+from typing import cast
+
+SMTP_PORT = 1025
+API_PORT = 8025
+
+
+@pytest.fixture(scope="module")
+def fake_smtp_server():
+    """Start fake SMTP server container"""
+    with DockerContainer("mailhog/mailhog") \
+            .with_exposed_ports(SMTP_PORT, API_PORT) as container:
+
+        smtp_port = container.get_exposed_port(SMTP_PORT)
+        api_port = container.get_exposed_port(API_PORT)
+        smtp_host = container.get_container_host_ip()
+        api_base = f"http://{smtp_host}:{api_port}"
+
+        for _ in range(10):
+            try:
+                requests.get(f"{api_base}/api/v2/messages")
+                break
+            except requests.ConnectionError:
+                import time
+                time.sleep(1)
```

Comment:
Ah, an interesting way to make sure the container is running!

TestContainers provide a similar feature already. Check the `DockerContainer` source for an example. You can give it an expected string. Once that string appears in the logs, then the container is considered ready to be used. You can also find a example at `test/integration/test_paramiko_platform.py`. Just check what the container used here outputs when it's ready to receive requests.

With that I think we can remove the `requests` dependency from this test :+1: